### PR TITLE
add a check for undefined array

### PR DESCRIPTION
**Pull request by SovereignAndrey**
Ticket(s):
-

## Explanation of the solution
Cards in Marketplace & Profiles details col logic wasn't handling possiblilty of undefined `active_in_countries` array. Adds that check in before `isEmpty()` check. 

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

File: src/pages/Profile/Body/GeneralInfo/DetailsColumn/Details.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default function Details() {
         <Detail title="address">{profile.street_address}</Detail>
       )}
       <Detail title="active in">
-        {isEmpty(profile.active_in_countries)
+        {!!profile.active_in_countries && isEmpty(profile.active_in_countries)
```
Comment (misicnenad) [inline src/pages/Profile/Body/GeneralInfo/DetailsColumn/Details.tsx:23]:
```suggestion
        {!profile.active_in_countries || isEmpty(profile.active_in_countries)
```

Comment (misicnenad) [inline src/pages/Profile/Body/GeneralInfo/DetailsColumn/Details.tsx:23]:
```suggestion
        {!profile.active_in_countries || isEmpty(profile.active_in_countries)
```

Comment (misicnenad) [inline src/pages/Profile/Body/GeneralInfo/DetailsColumn/Details.tsx:23]:
```suggestion
        {!profile.active_in_countries || isEmpty(profile.active_in_countries)
```

**Review by misicnenad — CHANGES_REQUESTED** [review]

File: src/pages/Profile/Body/GeneralInfo/DetailsColumn/Details.tsx (hunk 1 above)
Comment (misicnenad) [reply src/pages/Profile/Body/GeneralInfo/DetailsColumn/Details.tsx:23]: If `active_in_countries` is undefined OR it is empty, show `HQ Country`.
If `active_in_countries` is defined and non-empty, show the list.

File: src/pages/Marketplace/Cards/Card.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ export default function Card({
             </p>
             <p className="line-clamp-2">
               <span className="font-semibold">Active in:</span>{" "}
-              {isEmpty(active_in_countries)
+              {!active_in_countries || isEmpty(active_in_countries)
```
Comment (ap-justin) [inline src/pages/Marketplace/Cards/Card.tsx:63]: @SovereignAndrey, should first update `type EndowmentBase` in `src/types/aws/ap/index.ts` so not to miss other fields

```ts
type EndowmentBase = {
  //...
  active_in_countries?: string[];
}
```

**Review by ap-justin — CHANGES_REQUESTED** [review]

**Comment by SovereignAndrey** [comment]
Closing for: https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1809

**Pull request closed without merging**